src/roswire/ros2/state.py

```diff
@@ -14,12 +18,53 @@
 class ROS2StateProbe:
     """Provides an interface for obtaining the instantaneous state of a ROS
     system in terms of its publishers, subscribers, and services."""
+    _app_instance: 'AppInstance' = attr.ib()
+
     @classmethod
     def for_app_instance(cls, app_instance: 'AppInstance') -> 'ROS2StateProbe':
-        raise NotImplementedError
+        return ROS2StateProbe(app_instance=app_instance)
 
     def probe(self) -> SystemState:
         """Obtains the instantaneous state of the associated ROS system."""
-        raise NotImplementedError
+        mode_dict: Dict[str, Dict[str, List[str]]] = {'pub': {},
+                                                      'sub': {},
+                                                      'serv': {}}
+        command = "rosnode list"
+        try:
+            output = self._app_instance.shell.check_output(command, text=True)
+        except dockerblade.exceptions.CalledProcessError:
+            logger.debug("Unable to retrieve rosnode list from command line")
```

Comment:
This `except` block is missing a `raise`. Right now, the program will write to the log and then continue executing the rest of the method.